Packages/Tracking/Physical Hands/Runtime/Scripts/PhysicalHandsButtonBase.cs

```diff
@@ -0,0 +1,407 @@
+        private ConfigurableJoint _configurableJoint;
+        private Vector3 _initialButtonPosition = Vector3.zero;
+        private List<GameObject> _objectsContactingButton = new List<GameObject>();
+
+        protected PhysicalHandsButtonHelper _buttonHelper;
+
+        [SerializeField]
+        public UnityEvent OnButtonPressed;
+        [SerializeField]
+        public UnityEvent OnButtonUnPressed;
+        [SerializeField]
+        public UnityEvent<ContactHand> OnHandContact;
+        [SerializeField]
+        public UnityEvent<ContactHand> OnHandContactExit;
+        [SerializeField]
+        public UnityEvent<ContactHand> OnHandHover;
+        [SerializeField]
+        public UnityEvent<ContactHand> OnHandHoverExit;
+
+#if UNITY_EDITOR
+        public void UpdateDistanceValues()
+        {
+            if (!Application.isPlaying)
+            {
+                if (_automaticOffsetDistance)
+                {
+                    if (_pressableObject != null &&
+                        _pressableObject.GetComponent<MeshFilter>() &&
+                        GetComponent<MeshFilter>())
+                    {
+                        _buttonTravelOffset = GetComponent<MeshFilter>().sharedMesh.bounds.extents.y;
+                    }
+                    else if (UltraleapSettings.Instance.showPhysicalHandsButtonOffsetWarning == true)
+                    {
+                        if (EditorUtility.DisplayDialog("Could not automatically set offset distance.",
+                            "Could not automatically set offset distance. \n \n Please either add mesh filters to both pressable object and physical hands button or manually set the offset.",
+                            "Do Not Show Again", "Okay"))
+                        {
+                            UltraleapSettings.Instance.showPhysicalHandsButtonOffsetWarning = false;
+                        }
+                        _automaticOffsetDistance = false;
+                    }
+                    else
+                    {
+                        _automaticOffsetDistance = false;
+                    }
+                }
+                if (_automaticTravelDistance)
+                {
+                    // leave 1mm so the button does not clip into the base obect (assuming same thickness)
+                    _buttonTravelDistance = Mathf.Abs(_pressableObject.transform.localPosition.y) - (_buttonTravelOffset);
+                }
+
+            }
+        }
+#endif
+
+        private void Start()
+        {
+            Initialize();
+        }
+
+        private void OnDestroy()
+        {
+            _buttonHelper._onHandContact -= OnHandContactPO;
+            _buttonHelper._onHandContactExit -= OnHandContactExitPO;
+            _buttonHelper._onHandHover -= OnHandHoverPO;
+            _buttonHelper._onHandHoverExit -= OnHandHoverExitPO;
+            _buttonHelper._onCollisionEnter -= OnCollisionPO;
+            _buttonHelper._onCollisionExit -= OnCollisionExitPO;
```

Comment:
these should include null checks as the _buttonHelper can be null when the whole button is destroying

Also this probably isn't needed as we unsub before we sub on initialize anyway